contracts/5/IbETHRouter.sol

```diff
@@ -0,0 +1,301 @@
+    }
+
+    // Remove liquidity from ibETH-Alpha Pool and convert all ibETH to Alpha 
+    // 1. Remove ibETH and Alpha from the pool.
+    // 2. Swap ibETH for Alpha.
+    // 3. Return Alpha to caller.   
+    function removeLiquidityAllAlpha(        
+        uint256 liquidity,
+        uint256 amountAlphaMin,        
+        address to,
+        uint256 deadline
+    ) public returns (uint256 amountAlpha) {                  
+        TransferHelper.safeTransferFrom(lpToken, msg.sender, address(this), liquidity);          
+        (uint256 removeAmountAlpha, uint256 removeAmountIbETH) = IUniswapV2Router02(router).removeLiquidity(
+            alpha,
+            ibETH,
+            liquidity,
+            0,
+            0,
+            address(this),
+            deadline
+        );        
+        address[] memory path = new address[](2);
+        path[0] = ibETH;
+        path[1] = alpha;
+        uint256[] memory amounts = IUniswapV2Router02(router).swapExactTokensForTokens(removeAmountIbETH, 0, path, to, deadline);               
+        TransferHelper.safeTransfer(alpha, to, removeAmountAlpha);                        
+        amountAlpha = removeAmountAlpha.add(amounts[1]);
+        require(amountAlpha >= amountAlphaMin, "IbETHRouter: receive less Alpha than amountAlphaMin");                               
+    }       
+
+    // Swap exact amount of ETH for Token
+    // 1. Receive ETH from caller
+    // 2. Wrap ETH to ibETH.
+    // 3. Swap ibETH for Token    
+    function swapExactETHForAlpha(
+        uint256 amountAlphaOutMin,        
+        address to,
+        uint256 deadline
+    ) external payable returns (uint256[] memory amounts) {                           
+        IBank(ibETH).deposit.value(msg.value)();   
+        address[] memory path = new address[](2);
+        path[0] = ibETH;
+        path[1] = alpha;     
+        uint256[] memory swapAmounts = IUniswapV2Router02(router).swapExactTokensForTokens(IBank(ibETH).balanceOf(address(this)), amountAlphaOutMin, path, to, deadline);
+        amounts = new uint256[](2);        
+        amounts[0] = msg.value;
+        amounts[1] = swapAmounts[1];
+    }
+
+    // Swap Token for exact amount of ETH
+    // 1. Receive Token from caller
+    // 2. Swap Token for ibETH.
+    // 3. Unwrap ibETH to ETH.
+    function swapAlphaForExactETH(
+        uint256 amountETHOut,
+        uint256 amountAlphaInMax,         
+        address to,
+        uint256 deadline
+    ) external returns (uint256[] memory amounts) {
+        TransferHelper.safeTransferFrom(alpha, msg.sender, address(this), amountAlphaInMax);
+        address[] memory path = new address[](2);
+        path[0] = alpha;
+        path[1] = ibETH;
+        uint256[] memory swapAmounts = IUniswapV2Router02(router).swapTokensForExactTokens(exactETHToIbETH(amountETHOut), amountAlphaInMax, path, address(this), deadline);                           
```

Comment:
Don't think this is correct. Shouldn't this use `ibETHForExactETH`? Alternatively we can remove this function for now as I don't think it's needed for v1 frontend?

Reply:
Yes this should use `ibETHForExactETH`. Add the function.